src/Workspaces/CoreTest/SymbolKeyTests.cs

```diff
@@ -647,6 +647,74 @@ object LocalFunction<T>()
             Assert.True(tested);
         }
 
+        [Fact, WorkItem(17702, "https://github.com/dotnet/roslyn/issues/17702")]
+        public void TestTupleWithLocalTypeReferences1()
+        {
+            var source = @"
+using System.Linq;
+
+class C
+{
+    void Method((C, int) t)
+    {
+    }
+}";
+            // Tuples store locations along with them.  But we can only recover those locations
```

Comment:
Great point jason.  I've updated this PR because of this.  Now, location resolution does not happen by default.  Instead, the client of SymbolKey has to opt-into this if they're in a scenario where they know this is safe (i.e. that the compilations should be exactly the same).  OOP-symbol-results are an example of where we can use this safely.